src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -150,7 +150,7 @@ private void initializeProducer() throws Exception {
   public void start() {
     if (_running.compareAndSet(false, true)) {
       initializeStateForPartitions();
-      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 40_000, 40_000, TimeUnit.MILLISECONDS);
+      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 1000, 30000, TimeUnit.MILLISECONDS);
```

Comment:
I am not sure if I fully understand your comment. Are you concerned with reducing delay of the first execution from 40 sec to 1 sec? Or are you concerned with reducing the periodic delay from 40 sec to 30 sec? Can you explain a bit more what strange thing may happen with shorter delay?